### PR TITLE
Feature/mobile view

### PR DESCRIPTION
**Pull request by DemogorGod**
Hey guys upon @natemiller1 request I have created a mobile view that encapsulates everything in the app that we have created, I know that we are adding other things in the app and i am willing to change this as we add more views and concepts into the app as well. Thanks! Enjoy!

Some kinks to workout with @ccali11  figuring out the user flow as on connects a wallet to sign in (which tab to follow to and how it connects) other kinks to work out as the flow gets ironed out. 

https://github.com/consensusnetworks/casimir/assets/63758598/5f4c0846-9a77-4b95-beff-4d5264ab35df

### CODE REVIEW

**Comment by DemogorGod** [comment]
@ccali11 @shanejearley 

there is a conflic in `contracts/ethereum/docs/index.md` I will let that be handled by you guys so i do not mess it up, Thanks!

File: apps/web/src/App.vue [hunk 1]
```diff
@@ -1,22 +1,53 @@
...
+    } else {
+      showMobileView.value = false
+    }
+  })
+
   onMounted(() => {
+    findScreenDimenstions()
+
+    window.addEventListener('resize', findScreenDimenstions)
+    window.addEventListener('orientationchange', findScreenDimenstions)
```
Comment (ccali11) [inline apps/web/src/App.vue:37]: camelCase `orientationChange`?

File: apps/web/src/App.vue [hunk 2]
```diff
@@ -1,22 +1,53 @@
 <script lang="ts" setup>
   import { RouterView } from 'vue-router'
   import DefaultLayout from '@/layouts/default-layout.vue'
+  import MobileLayout from '@/layouts/mobile-layout.vue'
```
Comment (ccali11) [inline apps/web/src/App.vue:4]: Ok cool. So you went with the mobile layout approach. 👍

File: apps/web/src/components/mobileComponents/MobileBreakdown.vue [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    })
+  }
+}
+
+onMounted(async () => {
+  if (user.value?.id) {
+    await getUserAnalytics()
+    setChartData()
+    await refreshBreakdown()
+    // TODO: Potentially find a better place to initialize these listeners
```
Comment (ccali11) [inline apps/web/src/components/mobileComponents/MobileBreakdown.vue:85]: We can resolve this in a separate PR

File: apps/web/src/components/mobileComponents/MobileConnectWallets.vue [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  selectProvider,
+} = useWallet()
+
+const authFlowCardNumber = ref(1)
+
+const selectedProivder = ref(null as null | string)
+
+const termsCheckbox = ref(true)
+const showTermsOfService = ref(false)
+const convertString = (inputString: string) => {
```
Comment (ccali11) [inline apps/web/src/components/mobileComponents/MobileConnectWallets.vue:21]: Rename to something like `abbreviateAddress`. This is also probably used several places throughout the app; should we have a `useHelpers` composable or something like that?

File: apps/web/src/layouts/mobile-layout.vue [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+const mobileViews = ref({
+    wallets: MobileConnectWallets,
+    breakdown: MobileBreakdown,
+    stake: MobileStake,
+    accounts: MobileAccounts,
+    settings: MobileSetting
+})
+
+onMounted(() => {
+  setTimeout(() =>{
```
Comment (ccali11) [inline apps/web/src/layouts/mobile-layout.vue:27]: why `setTimeout`?

File: apps/web/src/components/mobileComponents/MobileBreakdown.vue [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+import { ref, onMounted, watch } from 'vue'
+import useContracts from '@/composables/contracts'
+import useUsers from '@/composables/users'
+import useEthers from '@/composables/ethers'
+import { AnalyticsData, ProviderString } from '@casimir/types'
+
+const { currentStaked, listenForContractEvents, refreshBreakdown, stakingRewards, totalWalletBalance } = useContracts()
+const { user, getUserAnalytics, userAnalytics } = useUsers()
+const { listenForTransactions } = useEthers()
+
```
Comment (ccali11) [inline apps/web/src/components/mobileComponents/MobileBreakdown.vue:12]: Remove space

File: apps/web/src/layouts/mobile-layout.vue [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+import { ref, onMounted } from 'vue'
+import useUser from '@/composables/users'
+import VueFeather from 'vue-feather'
+import MobileConnectWallets from '../components/mobileComponents/MobileConnectWallets.vue'
+import MobileBreakdown from '../components/mobileComponents/MobileBreakdown.vue'
+import MobileStake from '../components/mobileComponents/MobileStake.vue'
+import MobileAccounts from '../components/mobileComponents/MobileAccounts.vue'
+import MobileSetting from '../components/mobileComponents/MobileSetting.vue'
+
+
```
Comment (ccali11) [inline apps/web/src/layouts/mobile-layout.vue:11]: Remove space

**Review by ccali11 — CHANGES_REQUESTED** [review]
@DemogorGod - Awesome work. App looks great. I prefer icon/navigation tray to be on the bottom of the screen, but I imagine you tried that first. Might be worth revisiting and figuring it out (in a later PR) unless you have good examples of other familiar apps with good UX that use that design approach. 

I made a couple small comments with tiny tweaks/suggestions. Take a look and let me know and then I will approve.

**Comment by ccali11** [comment]
And @shanejearley - I pulled down to see what the changes were to the file with conflicts; I'm not sure which process is managing the `contracts/ethereum/docs/index.md`, but I suppose there are what looks like some benign changes. Will let you address.

**Comment by shanejearley** [comment]
TS errors need fixing.

File: apps/web/src/layouts/mobile-layout.vue (hunk 7 above)
Comment (shanejearley) [reply apps/web/src/layouts/mobile-layout.vue:11]: Resolved.

**Comment by shanejearley** [comment]
@DemogorGod make sure to pull and install before making new commits to respond to Chris's requests. Conflicts and TS errors are resolved.

**Comment by DemogorGod** [comment]
upon request from the ui designer, this branch will be deleted and we will move away from the mobile layout.

**Pull request closed without merging**